### PR TITLE
Cleaned up some pep8 errors and fixes #3167 

### LINKED ISSUE #3167: "Jeffreys" misspelled as "Jeffrey's"
The documentation for [`statsmodels.stats.proportion.proportion_confint`](http://statsmodels.sourceforge.net/devel/generated/statsmodels.stats.proportion.proportion_confint.html) describes an option as "Jeffrey’s Bayesian Interval". The interval in question is named after the Jeffreys prior, which itself is named after Harold Jeffreys. The option itself should then be named `jeffreys`, not `jeffrey`.

### PR DESCRIPTION
**Pull request by jvhaggard**
STY: Small spacing cleanups
BUG: Fixed typo in option #3167

### CODE REVIEW

File: statsmodels/stats/proportion.py [hunk 1]
```diff
@@ -112,14 +112,21 @@ def func(qi):
         ci_low = center - dist
         ci_upp = center + dist
 
+    # This is a typo cleaned up in issue #3167 and will need to be deprecated in
+    # a future release.
     elif method == 'jeffrey':
```
Comment (josef-pkt) [inline statsmodels/stats/proportion.py]: replace this by `method[:4] == 'jeff'`
We started to switch to just checking unambigous part of the string in a few methods. This avoids deprecation and more forgiving to users that misspell the name.

**Comment by josef-pkt** [comment]
@jvhaggard  Thank you

you could amend the commit and force push with the change to `method[:4] == 'jeff'` (or squash)
Then, I think it's ready to merge, either tonight or tomorrow.

File: statsmodels/stats/proportion.py [hunk 2]
```diff
@@ -112,14 +112,16 @@ def func(qi):
         ci_low = center - dist
         ci_upp = center + dist
 
-    elif method == 'jeffrey':
-        ci_low, ci_upp = stats.beta.interval(1 - alpha,  count + 0.5,
-                                             nobs - count + 0.5)
+    # method adjusted to be more forgiving of misspellings or incorrect option name
+    elif method[:4] == 'jeff':
+            ci_low, ci_upp = stats.beta.interval(1 - alpha, count + 0.5,
```
Comment (josef-pkt) [inline statsmodels/stats/proportion.py]: do you have a `tab` in there instead of 4 spaces?
this shows a double intendation

Comment (jvhaggard) [inline statsmodels/stats/proportion.py]: You are correct, I had an extra 4 spaces in there.  I must have hit tab accidentally.

**Comment by josef-pkt** [comment]
@jvhaggard  Thank you, 

looks good now, I will merge when travis comes back green

**Pull request merged**